### PR TITLE
vimPlugins.render-markdown-nvim: add nvim-treesitter dependency

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

From their [README](https://github.com/MeanderingProgrammer/render-markdown.nvim?tab=readme-ov-file#install)
it seems that treesitter is required.
The rest look like optional dependencies.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 343557` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>vimPlugins.render-markdown-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Comment by teto** [comment]
it should not be too hard to test the plugin. add the dependencies into RTP (since nixpkgs infra doesn't do that yet) , open a markdown and see if if neovim errors out. like 
`nvim -c"set rtp+=${vimPlugins.nvim-treesitter}" -c "${render-markdown-nvim}" "PreviewMarkdown" -es  +quitall! `

Some tests are visible for reference at
https://github.com/NixOS/nixpkgs/pull/343769

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 1]
```diff
@@ -1498,6 +1498,10 @@
     dependencies = with self; [ nvim-treesitter plenary-nvim ];
   };
 
+  render-markdown-nvim = super.render-markdown-nvim.overrideAttrs {
+    dependencies = with self; [ nvim-treesitter ];
```
Comment (PerchunPak) [inline pkgs/applications/editors/vim/plugins/overrides.nix:1502]: `nvimRequireCheck = "render-markdown"`?

**Review by PerchunPak — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/vim/plugins/overrides.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/applications/editors/vim/plugins/overrides.nix:1502]: Sure ! Done.

**Review by PerchunPak — APPROVED** [review]
LGTM

**Comment by GaetanLepage** [comment]
> it should not be too hard to test the plugin. add the dependencies into RTP (since nixpkgs infra doesn't do that yet) , open a markdown and see if if neovim errors out.

This didn't make it to the PR in the end.
Do you have a plugin examples where those tests are used already ?

**Comment by mrcjkb** [comment]
this is one of those cases where the plugin doesn't actually depend on nvim-treesitter, but it [requires one or multiple parsers to be installed](https://github.com/MeanderingProgrammer/render-markdown.nvim/blob/75a0a9596a91130fae43d3b7c0d6c651645ef1df/.github/workflows/luarocks.yml#L37).

In such cases, adding nvim-treesitter as a dependency won't make the plugin work out of the box. Users will still have to specify `nvim-treesitter.withPlugins ...`

**Comment by GaetanLepage** [comment]
> This is one of those cases where the plugin doesn't actually depend on nvim-treesitter, but it [requires one or multiple parsers to be installed](https://github.com/MeanderingProgrammer/render-markdown.nvim/blob/75a0a9596a91130fae43d3b7c0d6c651645ef1df/.github/workflows/luarocks.yml#L37).
> 
> In such cases, adding nvim-treesitter as a dependency won't make the plugin work out of the box. Users will still have to specify `nvim-treesitter.withPlugins ...` There's no need to add nvim-treesitter as a dependency if the plugin doesn't actually try to `require` one of its [(legacy)](https://github.com/nvim-treesitter/nvim-treesitter/issues/4767) modules.

Right, I forgot about this. I will revert it to prevent such noise from remaining in the code base.

**Pull request merged**